cytotable/convert.py

```diff
@@ -1014,6 +1116,7 @@ def convert(  # pylint: disable=too-many-arguments,too-many-locals
     ),
     infer_common_schema: bool = True,
     drop_null: bool = True,
+    data_type_cast_map: Optional[Dict[str, str]] = None,
```

Comment:
Good thinking; I could go either way here. My impression was that casting the data would have additional performance and time costs so felt a default of None might be apt. Maybe aligning with #56's resolution would be wise here; avoiding data changes unless provided.

Reply:
I agree setting low storage defaults is a good approach. I'm also wondering why we need to cast in the first place? See other, more specific comments throughout, but why not output the parquet files initially with the specified data types? (instead of having to read/write/cast again)

Reply:
Good questions @gwaybio thank you! We need to cast because we are converting to different data sources and each data type change involves embedded inferenced data typing from source to target. For example, SQLite uses storage class data types which do not perfectly align with Arrow data types so this conversion is inferenced to safely move data from one format to the next (through DuckDB currently). Adding the defaults here I feel are necessary in terms of implementing a Cytomining opinion about the data and expected results downstream. Without them we are leaving those decisions up to whichever query client we use (for example Pandas, Arrow, and others build this into their queries by default) and this will normally include larger rather than smaller data types to lean towards (I believe) data retention and performance (for example, preferring float64 over float32).

To resolve this at data read instead of rewriting the files I think we will need to accomplish #67 . Addressing this would mean capturing the column names and types, then using SQL to cast the values individually as part of the extraction queries. This will involve a fairly different approach and refactoring to this PR (needing a full re-review or new PR altogether).

Reply:
> To resolve this at data read instead of rewriting the files 

I might be a bit confused.

Right now I interpret what is happening as `read SQLite` -> `write parquet` -> `cast dtypes` -> `overwrite parquet`. Is this not what is happening?

Reply:
i.e. why not solve at the first data write (not the overwrite)?

Reply:
Your description is accurate. My hope is to resolve at the SQLite (or other data type source) read step meaning the data in-memory before initial write follows the memory cost of what will eventually be initially written (and removing the need for rewrite).

Reply:
Double checking here: should we move to a `convert.data_type_cast_map` default of: `{"integer": "int32", "float": "float32",}`? My only hesitation would be a gap / misunderstanding on a user's part concerning data loss (if they didn't witness or use a non-None parameter).

Reply:
IIRC, we had talked about keeping 64 the default, and providing a 32 as an optional argument that is "experimental but will decrease storage requirements"